app/src/main/java/org/mozilla/fenix/utils/Undo.kt

```diff
@@ -40,29 +45,60 @@ fun CoroutineScope.allowUndo(
     // writing a volatile variable.
     val requestedUndo = AtomicBoolean(false)
 
-    // Launch an indefinite snackbar.
-    val snackbar = FenixSnackbar
-        .make(view, FenixSnackbar.LENGTH_INDEFINITE)
-        .setText(message)
-        .setAnchorView(anchorView)
-        .setAction(undoActionTitle) {
-            requestedUndo.set(true)
+    // Launch an alert dialog for undo- for accessibility users
+    fun showUndoDialog() {
+        val dialogBuilder = AlertDialog.Builder(view.context)
+        dialogBuilder.setMessage(message).setCancelable(false).setPositiveButton(R.string.a11y_dialog_deleted_confirm) { _, _ ->
+            launch {
+                operation.invoke()
+            }
+        }.setNegativeButton(R.string.a11y_dialog_deleted_undo){_,_ ->
             launch {
                 onCancel.invoke()
             }
         }
+        val alert = dialogBuilder.create()
+        alert.show()
+    }
 
-    // If user engages with the snackbar, it'll get automatically dismissed.
-    snackbar.show()
+    // Launch an indefinite snackbar
+    fun showUndoSnackbar() {
+        val snackbar = FenixSnackbar
+            .make(view, FenixSnackbar.LENGTH_INDEFINITE)
+            .setText(message)
+            .setAnchorView(anchorView)
+            .setAction(undoActionTitle) {
+                requestedUndo.set(true)
+                launch {
+                    onCancel.invoke()
+                }
+            }
+
+        // If user engages with the snackbar, it'll get automatically dismissed.
+        snackbar.show()
 
-    // Wait a bit, and if user didn't request cancellation, proceed with
-    // requested operation and hide the snackbar.
-    launch {
-        delay(UNDO_DELAY)
+        // Wait a bit, and if user didn't request cancellation, proceed with
+        // requested operation and hide the snackbar.
+        launch {
+            delay(UNDO_DELAY)
 
-        if (!requestedUndo.get()) {
-            snackbar.dismiss()
-            operation.invoke()
+            if (!requestedUndo.get()) {
+                snackbar.dismiss()
+                operation.invoke()
+            }
         }
     }
+
+    // Check if accessibility is enabled
+    val isAccessibilityEnabled = accessibilityEnabled(view)
+
+    // Show either Snackbar or dialog based on result
+    if (isAccessibilityEnabled) showUndoDialog() else showUndoSnackbar()
+}
+
+
+
+fun accessibilityEnabled(view:View): Boolean {
+    val am = view.context.getSystemService(Context.ACCESSIBILITY_SERVICE) as AccessibilityManager
+    return am.isEnabled
```

Comment:
Good helper function, this really improves the clarity.

Normally I think it's good to narrow parameters as much as possible (passing `Context` is better than `View`, `AccessibilityManager` is better than `Context`).  There are a lot of good reasons for doing that, but in this instance the lookup is all the function is really doing.  So this is probably good as is!  No changes needed.